Sources/MessagingPushAPN/MessagingPush+APN.swift

```diff
@@ -1,47 +1,50 @@
 import CioMessagingPush
 import CioTracking
 import Foundation
+#if canImport(UserNotifications)
+import UserNotifications
+#endif
 
-public protocol MessagingPushAPNInstance: AutoMockable {
-    // sourcery:Name=registerAPNDeviceToken
-    func registerDeviceToken(apnDeviceToken: Data)
-
-    // sourcery:Name=didRegisterForRemoteNotifications
-    func application(
-        _ application: Any,
-        didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
-    )
-
-    // sourcery:Name=didFailToRegisterForRemoteNotifications
-    func application(
-        _ application: Any,
-        didFailToRegisterForRemoteNotificationsWithError error: Error
-    )
-}
+// Expose `MessagingPush` in module so customers do not need to `import CioMessaginPush`
+// Only 1 import: `CioMessaginPushAPN`.
+public typealias MessagingPush = CioMessagingPush.MessagingPush
 
 /**
- MessagingPush extension to support APN push notification messaging.
+ Convenient extensions so singleton instances of `MessagingPush` can access functions from `MessagingPushAPN`.
   */
 extension MessagingPush: MessagingPushAPNInstance {
```

Comment:
For APN and FCM modules I refactored the `MessagingPush` extensions to not have any logic in them. They simply call the `MessagingPushX` class for the module. Because logic is moved away from the extensions, this makes code more testable.